pallets/auctions/src/types/candle.rs

```diff
@@ -0,0 +1,277 @@
+use super::*;
+
+///
+/// Implementation of Candle auction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>, Bid<T>> for CandleAuction<T> {
+	///
+	/// Creates a Candle Auction
+	///
+	fn create(&self, sender: T::AccountId, auction: &Auction<T>) -> DispatchResult {
+		self.validate_data()?;
+		Pallet::<T>::validate_create(&self.common_data)?;
+		Pallet::<T>::handle_create(sender, auction, &self.common_data)?;
+
+		Ok(())
+	}
+
+	///
+	/// Updates a Candle Auction
+	///
+	fn update(self, sender: T::AccountId, auction_id: T::AuctionId) -> DispatchResult {
+		self.validate_data()?;
+
+		<Auctions<T>>::try_mutate(auction_id, |maybe_auction| -> DispatchResult {
+			let auction_result = maybe_auction.as_mut().ok_or(Error::<T>::AuctionDoesNotExist)?;
+
+			if let Auction::Candle(candle_auction) = auction_result {
+				Pallet::<T>::validate_update(sender, &candle_auction.common_data)?;
+				*candle_auction = self;
+
+				Ok(())
+			} else {
+				Err(Error::<T>::NoChangeOfAuctionType.into())
+			}
+		})
+	}
+
+	///
+	/// Places a bid on an CandleAuction
+	///
+	fn bid(&mut self, auction_id: T::AuctionId, bidder: T::AccountId, bid: &Bid<T>) -> DispatchResult {
+		let closing_period_range = Pallet::<T>::determine_candle_closing_range(bid, self);
+		match closing_period_range {
+			Ok(range) => {
+				// <HighestBiddersByAuctionClosingRange<T>>::insert(&auction_id, &range, bidder.clone());
+				<HighestBiddersByAuctionClosingRange<T>>::mutate(
+					&auction_id,
+					&range,
+					|highest_bidder| -> DispatchResult {
+						*highest_bidder = Some(bidder.clone());
+
+						Ok(())
+					},
```

Comment:
Might be worth comparing this solution to one where you store a BTreeMap in one storage value. Will be more expensive on bid, but much cheaper on close, I imagine.

Also: I'm confused that you just set the `highest_bidder`, might be intentional, though (the details of candle auctions are wiped from my mind again :-D)